### ISSUE TITLE
WindowTop often doesn't work on discord

### ISSUE THREAD

**Issue by Vino4** [WindowTop/WindowTop-App#320]
Hello friends :D 

I love this product so much- it's literally exactly what I wanted for years so thank you so much <3

# Issue 
There's a bug which prevents me from putting discord into PiP mode

## Steps
- Open discord and login
- Get on a call with someone
- Have them share their screen
- Pop-out the shared screen
- Attempt to put any discord window in shared screen mode

### Expected
- You're able to hover at the center of the window and click PiP put the window in PiPmode
- OR you're able hover on the minimize button of the window click PiP to put the window in PIP more
- OR you're able to use the assigned keyboard shortcut to put the window in PiP mode

### Actual
- Hovering at the center of the window does not show the WindowTop toolbar
- Hovering on the minimize button does not show the WindowTop toolbar
- Using the keyboard shortcut does not put the window in PiPmode

### Notes
- While I am specifically interested in PiP, none of the other features of window top work on discord either
- All other windows still work fine and you're able to put them in PIP mode
- My OS is Windows 11 version `22H2`
- WindowTop version is `5.21.2`
- Discord version is `Stable 227102 (fe42a93)`
- Restarting WindowTop doesn't help
- Restarting discord doesn't help
- Restarting the computer *sometimes* helps
- *sometimes* this issue goes away and everything works, but more often than not it doesn't
- I'm not really sure about this but I have more of a chance to get the PIP mode to work if it's a shared screen in a server call rather than a direct call

**Comment by gileli121** [comment]
Thanks for the report.
I am trying to reproduce

**Comment by gileli121** [comment]
@Vino4 
I see that the app detects the window (you have a toolbar). At least the login window.

Do I have to do all the other steps?
Please make sure that the app is not excluded:

Go to
![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/1740c235-fb13-43c9-b24c-f3d88213d1af)

Make sure that Discord is not listed here
![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/30a2b87f-568f-4c41-8563-0474414866ee)

Or if it is listed, it can be listed only if "Whitelist the specified programs" is checked (and then WindowTop will ignore all other windows except this app and anything other than listed)

In addition, you may want to try v5.22-beta9 from here:


I made many changes and feature improvements for the PiP and the Crop in this version.

I noticed that beta 9 seems to be very stable now so it is probably one version before the final release of 5.22

**Comment by Vino4** [comment]
Thank you for being super responsive @gileli121 ^^

Discord is not in the list (it's empty) and the mode s set to "Blacklist the specified programs"
I logged out of discord and I can see the WindowTop bar on the login page and on the main discord app now- This issue seems to be specifically tied to something that happens when the other steps are done :)

I will download the beta and try it tomorrow when I have people around to test with :D

**Comment by Vino4** [comment]
@gileli121 Just a little update for you:
- The issue is indeed very specifically appears when you pop-out someone's shared screen, it starts on that particular screen!
- I have not had a chance to try the beta sadly, I'll try again tonight :)

**Comment by Vino4** [comment]
Additional info: I tried to shrink the subject window on 5.21.2 and got:
<img width="241" alt="image" src="https://github.com/WindowTop/WindowTop-App/assets/14026415/c995b607-790a-4b13-87ba-b965f78c0c3b">

**Comment by Vino4** [comment]
@gileli121 I just tested v5.22-beta9 beta and here are the results:
- Cropping the original window works great!
- I can see and use the hotbar on a popped out shared-screen window while on a server call
- I cannot see nor use the hotbar on a popped out shared-screen window while on a direct call

So as far as the hotbar issue goes, it's similar to v5.21.2 :)

**Comment by Vino4** [comment]
@gileli121 Now I'm seeing another issue (currently on v5.22-beta9)- when I crop a window that's being popped-out and shared on a discord server-call, the entire cropped window turns grey and uncropping it doesn't fix it:

<img width="1353" alt="image" src="https://github.com/WindowTop/WindowTop-App/assets/14026415/ed3018a8-e59c-4b6b-98c2-62ce623808bf">

**Comment by gileli121** [comment]
@Vino4 
Thanks, I hope to get to an investigation of it today.
In the last two days, I have been working on improving the hotkeys configuration, fixing some bugs, and adding improvements that allow you to edit the main toolbar hotkeys from the toolbar page. The part in green was added:

![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/e3392752-9a0d-4ab3-ae44-0c070185884f)

I think that I will start working on this bug today.

**Comment by gileli121** [comment]
I just reproduced it

Investigating

**Comment by gileli121** [comment]
This is the problem.

Because the title is ""
![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/8f1567f3-1239-46bc-a8d4-e65fc46cfe49)

I probably did it as some kind of bad workaround for something that later solved properly.

I think that need to remove the "" condition.

**Comment by gileli121** [comment]
Fixed

![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/8e96c65f-9a33-497b-88af-7c3cab873f43)

But I am wondering if somehow it can detect windows that it should ignore.
The fix is to allow empty titles

**Comment by gileli121** [comment]
Crop + PiP is working
![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/c9332c77-6791-4230-9fbe-ffa6cedef1b6)

But title is indeed empty.
I will add condition that if the title is empty, the PiP title will show the process name

**Comment by gileli121** [comment]
> But I am wondering if somehow it can detect windows that it should ignore.

Yes
![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/7625a57b-5d65-4a06-96d6-8d612951b29a)

**Comment by gileli121** [comment]
The simple fix was to keep using the "_WINDOWTOP_INTERNAL_UI_" title.
It was developed later but not used enough.
After adding it to the Anchors and other UI elements it works fine.

Now it does not show the arrow above anchors

**Comment by gileli121** [comment]
> I will add condition that if the title is empty, the PiP title will show the process name

Done.

![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/f5ac0b92-0117-4491-86bc-040c019c241c)

The same condition also used by its Anchor (same issue was also there)

**Comment by gileli121** [comment]
> @gileli121 Now I'm seeing another issue (currently on v5.22-beta9)
> 
> * When I crop a window that's being shared on a discord server-call (regardless of whether it's popped out or not), the entire cropped window turns grey
> * Uncropping it (if it's popped out) doesn't fix the video feed unless I pop it back in
> * Putting the window in PiP mode does the same thing
> * I've seen this happen in v5.21.2 only once when putting it PiP mode
> * This issue does **not** happen on other windows other than Discord's shared-screen window (example, a browser playing netflix)
> 
> <img alt="image" width="1353" src="https://user-images.githubusercontent.com/14026415/268427831-ed3018a8-e59c-4b6b-98c2-62ce623808bf.png">

Reproduced
![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/fbed7896-1838-47cd-8871-4bb7f4c3abc5)

**Comment by gileli121** [comment]
> > @gileli121 Now I'm seeing another issue (currently on v5.22-beta9)
> > 
> > * When I crop a window that's being shared on a discord server-call (regardless of whether it's popped out or not), the entire cropped window turns grey
> > * Uncropping it (if it's popped out) doesn't fix the video feed unless I pop it back in
> > * Putting the window in PiP mode does the same thing
> > * I've seen this happen in v5.21.2 only once when putting it PiP mode
> > * This issue does **not** happen on other windows other than Discord's shared-screen window (example, a browser playing netflix)
> > 
> > <img alt="image" width="1353" src="https://user-images.githubusercontent.com/14026415/268427831-ed3018a8-e59c-4b6b-98c2-62ce623808bf.png">
> 
> Reproduced ![image](https://user-images.githubusercontent.com/17680514/268530791-fbed7896-1838-47cd-8871-4bb7f4c3abc5.png)

Fixed.
Used the same hack from 
https://github.com/WindowTop/WindowTop-App/issues/146

![image](https://github.com/WindowTop/WindowTop-App/assets/17680514/37b1cc3d-5270-4875-8313-56768a5e4489)

**Comment by gileli121** [comment]
> Potentially helpful, additional info: I tried to shrink the subject window on 5.21.2 and got: <img alt="image" width="241" src="https://user-images.githubusercontent.com/14026415/268427002-c995b607-790a-4b13-87ba-b965f78c0c3b.png">

This possible may happen less due to a fix in recent betas of 5.22

> Improved app stability by removing almost all thread-locking mechanisms and replacing them with a less drastic approach. Freezing is now a very rare case

**Comment by gileli121** [comment]
@Vino4 

Please try beta 11 from here:

https://github.com/WindowTop/WindowTop-App/releases/tag/v5.22.0

Let me know if all issues are resolved.
And please try the INTERACT mode (you will see this button on the PiP view..)
I want to know if it works fine with it. If so, I will add it to the default apps that interact is enabled by default. 

The interact mode allows you to control the window via the PiP view.

**Comment by Vino4** [comment]
@gileli121 Thank you so much!!!!
I will test it tonight and report back 😁

**Comment by Vino4** [comment]
@gileli121 I've tested [v5.22.0](https://github.com/WindowTop/WindowTop-App/releases/tag/v5.22.0) and everything is amazing!!!
- Discord works well with PiP/Crop
- Crop works well with videos
- Interact mode is super cool, both smart drag and read mode are awesome!!

Once again thank you so much for your amazing software!! I sincerely appreciate it and I have been actively recommending it to everyone I know since the moment I discovered it last month because it is SO GOOD!!!😁

**Comment by gileli121** [comment]
@Vino4 
Great to hear!

WindowTop is a secret software.
Because I am nobody and not using the brand "Microsoft" (unlike PowerToys, which is also mostly developed by nobody people but uses the brand "Microsoft"...), you are not likely to hear about it every time a new feature is released. 
But this app's power is its quality level, and this is not my job to say. My job is to provide it. Users like you say it.
So, this app is 100% promoted by users themself. No blogs, no YouTube videos, nothing.
Only by users like you :) The business depends on users who spread the word. 

If you try to compare the crop here with what they did, you will probably laugh a lot, or you won't believe it, and this is the point - the fact that famous bloggers talk about something doesn't necessarily mean that the product is good. 

This world is made of fanboys, money, and interests so this is how it is.
Don't expect it to be promoted by tons of bloggers just because it is good like you saying. I know what I am saying. 

You and users like you need to spread the word :)
